### PR TITLE
Allow custom domains for inline damage rolls

### PR DESCRIPTION
**Pull request by CarlosFdez**
Restricted to sluggified that ends with -damage.

### CODE REVIEW

File: src/module/system/text-editor.ts [hunk 1]
```diff
@@ -676,7 +682,8 @@ async function augmentInlineDamageRoll(
             "inline-damage",
             item ? `${item.id}-inline-damage` : null,
             item ? `${sluggify(item.slug ?? item.name)}-inline-damage` : null,
-        ]);
+            args.domains,
```
Comment (stwlam) [inline src/module/system/text-editor.ts:685]: Can this be vetted?

Comment (CarlosFdez) [reply src/module/system/text-editor.ts:685]: Its vetted in the params, but we could move the point of validation to augmentDamageRoll() if you prefer.

File: src/module/system/text-editor.ts [hunk 2]
```diff
@@ -534,6 +536,7 @@ class TextEditorPF2e extends TextEditor {
 
         const item = args.rollData?.item instanceof ItemPF2e ? args.rollData?.item : null;
         const actor = (args.rollData?.actor instanceof ActorPF2e ? args.rollData?.actor : null) ?? item?.actor ?? null;
+        const domains = params.domains?.split(",").filter((d) => sluggify(d) === d && d.endsWith("-damage"));
```
Comment (stwlam) [inline src/module/system/text-editor.ts:539]:
```suggestion
        const domains = params.domains?.split(",")..map((d) => sluggify(d)).filter((d) =>  /^[a-z][-a-z0-9]+-damage$/.test(d));
```

Comment (CarlosFdez) [reply src/module/system/text-editor.ts:539]: We can do the regex but I think if its not a valid slug, it should fail rather than surprise anyone.

**Pull request merged**